File: examples/governance_static_go/domain/approved_model.go
package domain

import "time"

type ApprovedModel struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	ModelId string `json:"model_id"`
	Name string `json:"name"`
	Version string `json:"version"`
	ProviderId string `json:"provider_id"`
	Description string `json:"description"`
	RiskLevel string `json:"risk_level"`
	RegisteredAt time.Time `json:"registered_at"`
	ParentModelId string `json:"parent_model_id"`
	DerivationType string `json:"derivation_type"`
	Capabilities []Capability `json:"capabilities"`
	IntendedUses []IntendedUse `json:"intended_uses"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e ApprovedModel) EventName() string { return "ApprovedModel" }

func (e ApprovedModel) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/model_registry_domain/lib/model_registry_domain/ai_model/events/approved_model.rb
module ModelRegistryDomain
  class AiModel
    module Events
      class ApprovedModel
        attr_reader :aggregate_id, :model_id, :name, :version, :provider_id, :description, :risk_level, :registered_at, :parent_model_id, :derivation_type, :capabilities, :intended_uses, :status, :occurred_at

        def initialize(aggregate_id: nil, model_id: nil, name: nil, version: nil, provider_id: nil, description: nil, risk_level: nil, registered_at: nil, parent_model_id: nil, derivation_type: nil, capabilities: nil, intended_uses: nil, status: nil)
          @aggregate_id = aggregate_id
          @model_id = model_id
          @name = name
          @version = version
          @provider_id = provider_id
          @description = description
          @risk_level = risk_level
          @registered_at = registered_at
          @parent_model_id = parent_model_id
          @derivation_type = derivation_type
          @capabilities = capabilities
          @intended_uses = intended_uses
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
